instagram/src/components/PostContainer/PostsPage.js

```diff
@@ -0,0 +1,176 @@
+    if (parsedState) {
+      this.setState({
+        data: parsedState.data,
+        newCommentText: parsedState.newCommentText,
+        searchText: parsedState.searchText
+      });
+    } else if (!parsedState) {
+      // There might be a more succinct way to do this, but the following turns the timestamp string from dummyData into Unix Epoch Time (while leaving the other data alone). Earlier I had just edited dummy-data.js, so this is more elegant.
+      const data = dummyData.map(post => {
+        return (post = {
+          username: post.username,
+          visible: true,
+          thumbnailUrl: post.thumbnailUrl,
+          imageUrl: post.imageUrl,
+          likes: post.likes,
+          liked: false,
+          timestamp: moment(
+            post.timestamp,
+            "MMMM Do YYYY, h:mm:ss a"
+          ).valueOf(),
+          comments: post.comments,
+          newCommentText: ""
+        });
+      });
+      this.setState({
+        data: data,
+        searchText: ""
+      });
+    }
+  }
+
+  // Note: I think I have to call prevProps in order to call prevState, even though I don't use prevProps:
+  componentDidUpdate(prevProps, prevState) {
+    // Copies state into localStorage each time App.js is updated:
+    let storedState = this.state;
+    localStorage.setItem("storedState", JSON.stringify(storedState));
+    // I put the search function inside CDU so I could be sure that this.state.searchText is current before setting the visible property accordingly. Required conditional to avoid an infinite loop since I have CDU calling setState():
+    if (this.state.searchText !== prevState.searchText) {
+      // Once again, to get at the data nested in this.state, I need a spread operator:
+      let tempData = [...this.state.data];
+      // It was nice to use the humble forEach to solve a problem. Here forEach() checks whether post.username includes the current searchText, and updates post.visible as true or false accordingly:
+      tempData.forEach(post => {
+        if (!post.username.includes(this.state.searchText)) {
+          post.visible = false;
+        } else if (post.username.includes(this.state.searchText)) {
+          post.visible = true;
+        }
+      });
+      this.setState({
+        data: tempData
+      });
+    }
+  }
+
+  clearLocalStorage = e => {
+    e.preventDefault();
+    localStorage.clear();
+    window.location.reload();
+  };
+
+  changeHandler = e => {
+    e.preventDefault();
+    this.setState({
+      [e.target.name]: e.target.value
+    });
+  };
+
+  changeHandlerNested = (e, timestamp) => {
+    e.preventDefault();
+    const postIndex = this.state.data.findIndex(
+      post => post.timestamp === timestamp
+    );
+    let tempData = [...this.state.data];
+    // The following was very difficult to debug. Since [e.target.name] is being used as a property name here, why isn't it:
+    // tempData[postIndex].[e.target.name] (with a dot)?
```

Comment:
ah, this was something we talked about last week. just remember that object notation can be `either` bracket or dot notation, and while you can chain both types in the same expression you can't use both simultaneously. check out this article to further explain why bracket notation can generally be more useful in most cases (depending on the object in question): https://codeburst.io/javascript-quickie-dot-notation-vs-bracket-notation-333641c0f781